<launch>
    <!-- 是否保存点云PCD文件 -->
    <arg name="save_odom" default="true"/>
    <!-- 是否保存点云PCD文件 -->
    <arg name="save_pcd" default="true"/>
    <!-- pcd文件命名方式 0:序号命名 1:时间戳命名 -->
    <arg name="pcd_name_stamp" default="false"/>
    <!-- 是否逐帧保存点云文件 -->
    <arg name="frame_wise" default="true"/>
    <!-- 是否打印点云内容 -->
    <arg name="print_cloud" default="false"/>
    <!-- 雷达类型 ROBOSENSE = 1, VELODYNE=2, OUSTER=3,XYZI=4, XYZINORMAL=5 -->
    <arg name="lidar_type" default="4"/>
    <!-- Subscribed topics -->
    <!--雷达点云话题  -->
    <arg name="pointcloud_topic" default="/cloud_registered_body"/>
    <!-- 里程计话题 -->
    <arg name="odom_topic" default="/Odometry"/>

    <node pkg="test_tools" type="test_tools_node" name="test_tools"  output="screen" required="true">
        <param name="save_odom" type="bool" value="$(arg save_odom)"/>
        <param name="save_pcd" type="bool" value="$(arg save_pcd)"/>
        <param name="pcd_name_stamp" type="bool" value="$(arg pcd_name_stamp)"/>
        <param name="frame_wise" type="bool" value="$(arg frame_wise)"/>
        <param name="print_cloud" type="bool" value="$(arg print_cloud)"/>
        <param name="lidar_type" type="int" value="$(arg lidar_type)"/>
        <param name="pointcloud_topic" type="string" value= "$(arg pointcloud_topic)" />
        <param name="odom_topic" type="string" value= "$(arg odom_topic)" />
        <param name="outfile_path" type="string" value="$(find test_tools)/result/"/>
    </node>
    <!-- <node pkg="test_tools" type="set_default_param" name="set_default_param"  output="screen" required="true"/>
    <node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find test_tools)/launch/show.rviz" />
    <node name="reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/> -->
 </launch>
